azure/monitoring.tf

```diff
@@ -206,6 +218,13 @@ output "monitoring-password" {
   value     = random_password.monitoring-password.result
 }
 
+resource "azurerm_role_assignment" "private_dns_contributor" {
+  count                = var.private_dns_zone ? 1 : 0
+  scope                = module.networking.vnet_id
+  role_definition_name = "Network Contributor"
```

Comment:
Perhaps this needs to be `Private DNS Zone Contributor` based on comments from https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/kubernetes_cluster#private_dns_zone_id .  And the example: 
```
resource "azurerm_resource_group" "example" {
  name     = "example"
  location = "West Europe"
}

resource "azurerm_private_dns_zone" "example" {
  name                = "privatelink.eastus2.azmk8s.io"
  resource_group_name = azurerm_resource_group.example.name
}

resource "azurerm_user_assigned_identity" "example" {
  name                = "aks-example-identity"
  resource_group_name = azurerm_resource_group.example.name
  location            = azurerm_resource_group.example.location
}

resource "azurerm_role_assignment" "example" {
  scope                = azurerm_private_dns_zone.example.id
  role_definition_name = "Private DNS Zone Contributor"
  principal_id         = azurerm_user_assigned_identity.example.principal_id
}

resource "azurerm_kubernetes_cluster" "example" {
  name                    = "aksexamplewithprivatednszone1"
  location                = azurerm_resource_group.example.location
  resource_group_name     = azurerm_resource_group.example.name
  dns_prefix              = "aksexamplednsprefix1"
  private_cluster_enabled = true
  private_dns_zone_id     = azurerm_private_dns_zone.example.id

  # rest of configuration omitted for brevity

  depends_on = [
    azurerm_role_assignment.example,
  ]
}
```